main.js

```diff
@@ -379,40 +377,52 @@ function addWidgetsFromJSON(jsonObject, numberOfColumns, numberOfRows) {
-                const id = addCell();
-                makeHiddenCell(document.getElementById(id));
-                // console.log("HIDDEN KEY : " + key);
+
+        if (widget == undefined) {
+            // widget is not in the available list, hide
+            const id = addCell();
+            makeHiddenCell(document.getElementById("widget-box-" + id));
+        } else {
+            if (jsonObject.config[widget.config].widgetTitle !== undefined) {
+                name = jsonObject.config[widget.config].widgetTitle; 
+            } else if (jsonObject.config[widget.config].name !== undefined) {
+                name = jsonObject.config[widget.config].name; 
+            } else if (jsonObject.config[widget.config].title !== undefined) {
+                name = jsonObject.config[widget.config].title; 
+            } else if (widget.job !== undefined) {
+                name = widget.job; 
+            } else if (widget.widget !== undefined) {
+                name = widget.widget; 
             } else {
-                // console.log("LENGTH KEY : " + key);
+                name = "Name Missing";
             }
-            continue;
-        }
-        existingWidgets.push(widget);
-        // console.log("PUTTING KEY : " + key);
-
-        const widgetWidth = widget.width;
-        const widgetHeight = widget.height;
-        const widgetJob = widget.job;
-        const widgetWidget = widget.widget;
-        const widgetConfig = widget.config;
-
-        const newCellSize = [widgetWidth, widgetHeight];
-        const widgetID = addCell(newCellSize);
-
-        let selectedJob = getSelectListElement(jobsListName, widgetID);
-        let selectedWidget = getSelectListElement(widgetsListName, widgetID);
-
-        selectedJob.value = widgetJob;
-        selectedWidget.value = widgetWidget;
-
-        if (widgetConfig) {
-            const configValue = JSON.stringify(configurations[widgetConfig], null, '\t');
-            const configElement = document.getElementById(itemConfigTextName(widgetID));
-            if (configValue) {
-                configElement.value = configValue;
+        
+        
+            existingWidgets.push(widget);
+
+            const widgetWidth = widget.width;
+            const widgetHeight = widget.height;
+            const widgetWidget = widget.widget;
+            const widgetConfig = widget.config;
+
+            const newCellSize = [widgetWidth, widgetHeight];
+
+            const widgetID = addCell(newCellSize);
+
+            let widgetNameLabel = document.getElementById("name-label-" + widgetID);
+            let widgetConfigLabel = document.getElementById("config-label-" + widgetID);
+            let selectedWidget = getSelectListElement(widgetsListName, widgetID);
+
+            widgetNameLabel.innerText = name;
+            widgetConfigLabel.innerText = widgetConfig
+            selectedWidget.value = widgetWidget;
+            
+            if (widgetConfig) {
```

Comment:
On line 417, you're setting widgetConfigLabel to widgetConfig and then you're checking for it's existence here. Should 417 move into this if statemnt?

Reply:
Yes